### ISSUE TITLE
asdf.fish breaking PATH when there are spaces

### ISSUE THREAD

**Issue by losvedir** [asdf-vm/asdf#550]
### Steps to reproduce

in `.config/fish/config.fish` have just:

```
set PATH "/Applications/Visual Studio Code.app/Contents/Resources/app/bin" $PATH
```

in the terminal `echo $PATH[1]` (that is, first element of the $PATH list) and see `/Applications/Visual Studio Code.app/Contents/Resources/app/bin` returned in its entirety.

Now run `source ~/asdf/asdf.fish` and `echo $PATH[4]` (or whichever index is for the element past the shims added by the script), and see that it returns `/Applications/Visual` and that `echo $PATH[5]` returns `Studio`, and `echo $PATH[6]` returns `Code.app/Contents/Resources/app/bin`.

In other words, `asdf/asdf.fish` breaks the PATH when it includes a path with a space in it, by splitting it on the spaces and re-exporting each part as a separate path.

### Expected behavior

`source ~/asdf/asdf.fish` should not affect the `PATH` other than the shims it adds in front.

### Actual behavior

It breaks the PATH when there are spaces in the path of some element of it.

### Environment

**OS**: macOS Mojave 10.14.5

```
➜  ~ asdf --version
v0.7.3
➜  ~ fish --version
fish, version 3.0.2
```
